### PR TITLE
[Merged by Bors] - syncer/bootstrap: reduce log and add metrics for peer error

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation
- reduce logging in syncer and active set
- add metrics for peer error
- add thread-safety to data fetcher's atx sync cache

### CODE REVIEW

File: syncer/data_fetch.go [hunk 1]
```diff
@@ -377,23 +383,35 @@ func (d *DataFetch) receiveOpinions(ctx context.Context, req *opinionRequest, pe
...
-	peer := p2p.NoPeer
+func (d *DataFetch) pickAtxPeer(epoch types.EpochID) p2p.Peer {
+	d.mu.Lock()
+	defer d.mu.Unlock()
 	if _, ok := d.atxSynced[epoch]; !ok {
 		d.atxSynced[epoch] = map[p2p.Peer]struct{}{}
 		delete(d.atxSynced, epoch-1)
 	}
+	peers := d.fetcher.GetPeers()
 	for _, p := range peers {
```
Comment (poszu) [inline syncer/data_fetch.go]: Is it OK to hold the mutex locked for the duration of `d.fetcher.GetPeers()`? (IDK how long can it take to complete).

File: syncer/data_fetch.go [hunk 2]
```diff
@@ -377,23 +383,35 @@ func (d *DataFetch) receiveOpinions(ctx context.Context, req *opinionRequest, pe
...
+	d.atxSynced[epoch][peer] = struct{}{}
+}
+
+// GetEpochATXs fetches all ATXs published in the specified epoch from a peer.
+func (d *DataFetch) GetEpochATXs(ctx context.Context, epoch types.EpochID) error {
+	peers := d.fetcher.GetPeers()
+	if len(peers) == 0 {
+		return errNoPeers
+	}
+	peer := d.pickAtxPeer(epoch)
```
Comment (fasmat) [inline syncer/data_fetch.go]: This will call `d.fetcher.GetPeers()` again.

File: syncer/data_fetch.go (hunk 1 above)
Comment (countvonzero) [reply syncer/data_fetch.go]: thanks. the peer info is static info in p2p.
also only one sync can happen at at time. this lock is for future misuse.

File: syncer/data_fetch.go (hunk 2 above)
Comment (countvonzero) [reply syncer/data_fetch.go]: fixed. thanks

**Review by fasmat — APPROVED** [review]

**Review by poszu — APPROVED** [review]

**Comment by countvonzero** [comment]
bors merge

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**